### PR TITLE
SALTO-6964: avoid generating source maps where they are not needed 

### PR DESCRIPTION
**Pull request by ori-moisis**
Also, refactor dummy adapter a bit to allow it to handle scale better

---

_Additional context for reviewer_
Based on #6832 

---
_Release Notes_: 
Core:
- Small performance improvements by reducing source map calculations

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/cli/src/main.ts [hunk 1]
```diff
@@ -61,4 +61,6 @@ const main = async (): Promise<CliExitCode> => {
   }
 }
 // eslint-disable-next-line @typescript-eslint/no-floating-promises
-main().then(exitCode => process.exit(exitCode))
+main().then(exitCode => {
```
Comment (ori-moisis) [inline packages/cli/src/main.ts:64]: unrelated fix - when calling `process.exit` we prevent `beforeExit` hooks to do async work.
this causes some injected tools like [clinic's heap profiler](https://clinicjs.org/heapprofiler/) to not work properly

File: packages/dummy-adapter/src/generator.ts [hunk 2]
```diff
@@ -840,7 +847,11 @@ export const generateElements = async (
     }).flat()
   }
 
-  const generateUsersLike = ({ numOfUsers, numOfGroups, distributionFactor }: UsersGenerationParams): Element[] => {
+  const generateUsersLike = (): Element[] => {
```
Comment (ori-moisis) [inline packages/dummy-adapter/src/generator.ts:850]: unrelated change - align the interface of this generate function to be the same as the other ones

File: packages/dummy-adapter/src/generator.ts [hunk 3]
```diff
@@ -1102,28 +1114,30 @@ export const generateElements = async (
...
+  const allElements = defaultTypes
+    .concat(primitiveTypes)
+    .concat(types)
+    .concat(records)
+    .concat(objects)
+    .concat(profiles)
+    .concat(users)
+    .concat([new ObjectType({ elemID: new ElemID(DUMMY_ADAPTER, 'noPath'), fields: {} })])
+    .concat(envObjects)
+    .filter(e => !elementsToExclude.has(e.elemID.getFullName()))
```
Comment (ori-moisis) [inline packages/dummy-adapter/src/generator.ts:1131]: the reason for the change in the code here is to avoid the `...`

File: packages/core/src/app_config.ts [hunk 4]
```diff
@@ -125,7 +125,7 @@ const mergeConfigWithEnv = async (config: AppConfig): Promise<AppConfig> => {
 
 const configFromNaclFile = async (filepath: string): Promise<AppConfig> => {
   const buf = await readFile(filepath)
-  const configInstance = (await awu((await parse(buf, filepath)).elements).peek()) as InstanceElement
+  const configInstance = (await awu((await parse(buf, filepath, {}, false)).elements).peek()) as InstanceElement
```
Comment (Nurdok) [inline packages/core/src/app_config.ts:128]: IMO passing constants like this is forcing the reader to read the original signature, I suggest moving from a list of parameters to a single object parameter so this'll read as `functions: {}, calcSourceMap: false`.

Feel free to ignore.

**Review by Nurdok — APPROVED** [review]

File: packages/core/src/app_config.ts (hunk 4 above)
Comment (ori-moisis) [reply packages/core/src/app_config.ts:128]: I fully agree
this is an external interface, so changing it / changing its behavior I think requires a separate PR
